### ISSUE TITLE
Can download but cannot remove downloads

### ISSUE THREAD

**Issue by plottar** [SkyTubeTeam/SkyTube#957]
The subject pretty much explains it all:
I can download videos as much as I like but I can't delete them anymore. Neither from within Skytube (error message is: "Unable to delete downloaded file: '<path to file>'") nor with a file manager like Ghost Commander. But I can still remove videos that I downloaded before this strange behaviour has occured.

**Comment by gzsombor** [comment]
Yes, unfortunately Androids DownloadManager has a "couple" of bugs - different bugs for different Android versions.

Here I've collected  a couple of other ones: #529

**Comment by gzsombor** [comment]
Is it happens on external storage? Or where do you download these files?
Have you switched on the 'Separate folder for channel' flag?
I haven't seen a problem yet, where even a file manager couldn't delete the videos.

**Comment by plottar** [comment]
Thanks for your consideration, gzsombor.

Yes, since my internal storage is almost full, I set Skytube up to save videos to external storage (microSD card); I set the path to: /storage/8512-1C03/skytube_dls

The flag to "Download Videos Into Folders" is *not* set; every video is downloaded directly into the directory mentioned above.

Please note that the problem did not come up the moment when I changed the download directory to external; I did it about six months ago but the issue arose about four weeks ago...

**Comment by drogga** [comment]
@plottar - Android 11 ?

**Comment by plottar** [comment]
No, Android 6.0.1,  @drogga

**Comment by drogga** [comment]
@plottar - It would be nice if you edit your comment to include the other device/OS specifications & configuration details (stock, custom rom, rooted, magisk, custom recovery, ...).
I can't think of anything else, than write access issues (permission / owner, even tho FAT, FAT32 & exFAT filesystems don't actually support those, especially on Ext. Storage).
You can probably still delete them by attaching the MicroSD Card to another device, or even better - a Linux, Windows (or Mac) machine, but I see that this isn't the point ;)

**Comment by dagofr01** [comment]
Actually I have the same problem but I can delete it from external storage no problem with x-plore which I typically use. I have a Nokia 6 on A9 and a Nokia 6.1 on A10 each on stock FW, plain vanilla Android with those minor vendor adaptions.
It works on A9 but it fails on A10. And when I noticed it  I could not relate it to anything than a skytube change but I guess there were a few at that time may be. I cannot point to anything.

**Comment by dagofr01** [comment]
I have to add that the app was able to delete videos successfully before an update and then failed. So it does not miss any permission or so.
